### PR TITLE
Feature/116023 connect recurrence data 2

### PR DESCRIPTION
**Pull request by nealfennimore**
- Moves over logic to sagas
- Adds in syncRule action to prevent edit actions from looping

### CODE REVIEW

File: plugins/events-pro/src/modules/elements/from-time-range-picker/element.js [hunk 1]
```diff
@@ -21,94 +21,37 @@ const {
...
-	const { startTime } = stateProps;
-	const min = toSeconds( startTime, TIME_FORMAT_HH_MM );
-
-	if ( toSeconds( e.target.value, TIME_FORMAT_HH_MM ) <= min ) {
-		return;
-	}
 
-	dispatchProps.editRule( ownProps.index, { end_time: `${ e.target.value }:00` } );
+const onMultiDayChange = ( stateProps, dispatchProps, ownProps ) => ( e ) => {
+	dispatchProps.editRule( ownProps.index, { multi_day: !! e.target.value } );
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/elements/from-time-range-picker/element.js]: We should really be using the next constant keys here

File: plugins/events-pro/src/modules/data/blocks/recurring/selectors.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ export const getStartDate = createSelector(
 
 export const getStartTime = createSelector(
 	[ getRule ],
-	( rule ) => rule.start_time.slice( 0, -3 ),
+	( rule ) => rule.start_time
```
Comment (nealfennimore) [inline plugins/events-pro/src/modules/data/blocks/recurring/selectors.js]: @paulmskim 	this was causing errors so I took it out. There's some inconsistencies between the format used here and the time pickers

File: plugins/events-pro/src/modules/elements/exception-field/monthly.js [hunk 3]
```diff
@@ -26,11 +27,15 @@ const MonthlyField = ( { isMultiDay, typeOption, onTypeChange } ) => {
 				options={ options.EXCEPTION_OCCURRENCE_OPTIONS }
 				selected={ typeOption }
 				onChange={ onTypeChange }
+				blockType={ constants.EXCEPTION }
 			/>
 			<DayOfMonthPicker />
 			<FromTimeRangePicker />
 			{ isMultiDay && <RecurringToDateTimePicker /> }
-			<SeriesEnds rowLabel={ __( 'Exception ends', 'events-gutenberg' ) } />
+			<SeriesEnds
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/elements/exception-field/monthly.js]: this and all the `SeriesEnds` elements should also get an `index` prop

File: plugins/events-pro/src/modules/elements/from-time-range-picker/element.js [hunk 4]
```diff
@@ -21,94 +21,37 @@ const {
...
-	if ( isMultiDay ) {
-		const max = toSeconds( endTime, TIME_FORMAT_HH_MM );
+const onEndTimeChange = ( stateProps, dispatchProps, ownProps ) => ( e ) => {
+	const seconds = toSeconds( e.target.value, TIME_FORMAT_HH_MM );
+	const min = toSeconds( stateProps.startTime, TIME_FORMAT_HH_MM );
 
-		if ( toSeconds( e.target.value, TIME_FORMAT_HH_MM ) >= max ) {
-			return;
-		}
+	if ( seconds > min ) {
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/elements/from-time-range-picker/element.js]: This check should only happen if `multiDay` is `false`. So we have a few cases:
- If `multiDay = false` and if `seconds <= min`, then return, we won't update time
- If `multiDay = false` and if `seconds > min`, then update time.
- If `multiDay = true`, then update time, as there are no constraints.
We can do this in sagas, but we need to handle this logic somewhere.

File: plugins/events-pro/src/modules/elements/from-time-range-picker/element.js [hunk 5]
```diff
@@ -21,94 +21,37 @@ const {
...
+		dispatchProps.editRule( ownProps.index, { end_time: e.target.value } );
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/elements/from-time-range-picker/element.js]: The `TimePicker` input returns a time in `HH:MM` format, but the state needs to hold the time in `HH:MM:SS` format. We need to update this somewhere, whether it's sagas or here.

File: plugins/events-pro/src/modules/data/blocks/recurring/selectors.js [hunk 6]
```diff
@@ -44,7 +44,7 @@ export const getStartDate = createSelector(
 
 export const getStartTime = createSelector(
 	[ getRule ],
-	( rule ) => rule.start_time.slice( 0, -3 ),
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/data/blocks/recurring/selectors.js]: We need to either do this here or do it when we map state to props, as we're storing the time as `HH:MM:SS` format in the state but using it as `HH:MM` format in the `TimePicker` component.

**Pull request merged**